packages/mocha/README.md

```diff
@@ -1 +1,102 @@
 # @bigtest/mocha
+
+Convergent Mocha functions for testing against asynchronous states
+
+## Convergent Assertions
+
+Typically, when testing asynchronous states (such as rendered content
+in an application) your tests need to run at just the right moment so
+that they are executed in the correct context. If your tests run too
+soon, they will fail; if they run too slow, well then you just have
+slow tests.
+
+Converging on a state means asserting against a state until the
+assertion passes, in which case you have successfully converged on
+that state!
+
+This package uses
+[`@bigtest/convergence`](https://github.com/thefrontside/bigtest/tree/master/packages/convergence)
+to repeatedly run assertions until they pass, or until the timeout has
+almost expired. Performing tests in this way allows them to pass the
+moment the desired state is achieved. This results in very fast tests
+when testing asynchronous things.
+
+_Read the `@bigtest/convergence` [docs on
+convergences](https://github.com/thefrontside/bigtest/tree/master/packages/convergence#why-convergence)
+for more info as to why converging on a desired state is better than
+trying to time it properly._
+
+## What is this package?
+
+This package wraps Mocha's `it` in a convergence helper so that any
+assertions you write using `it` become convergent assertions that
+allow you to easily test asynchronous states.
+
+## Writing Tests
+
+Because convergent assertions are ran repeatedly until they pass, it
+is _highly recommended_ that you **do not** perform any side-effects
+in your assertions. This will result in the side-effect being ran
+multiple times in the case when the assertion fails.
+
+For this reason, you should keep your side-effect producing code in
+hooks and **out of your assertions.** These "pure assertions" also
+help your tests be more readable and explicit.
+
+#### :no_entry_sign: do not do this:
+
+``` javascript
+describe('my button', () => {
+  it('shows a loading indicator after clicking', () => {
+    // this will be called every time the assertion runs
+    $button.click();
+
+    // it might take a few milliseconds for any side-effects to
+    // happen, so this might fail the first time and cause this entire
+    // assertion to run again, thus clicking the button again
+    expect($button.className).to.include('is-loading');
+  });
+});
+```
+
+#### :white_check_mark: do this:
+
+``` javascript
+describe('clicking my button', () => {
+  // keep side-effects inside hooks
+  beforeEach(() => $button.click());
+
+  // a pure assertion has no side-effects; even if it fails, it can
+  // be run again and again without consequence
+  it('shows a loading indicator', () => {
+    expect($button.className).to.include('is-loading');
+  });
+});
+```
+
+### Convergent Hooks?
```

Comment:
If we want to phrase this as a question, then it has to connect directly to the reader's experience. E.g.

"What if I want to control the timing of my setup code?"

Otherwise, I'd just remove the question mark.